library/core/src/slice/sort/shared/smallsort.rs

```diff
@@ -834,9 +834,10 @@ unsafe fn bidirectional_merge<T: FreezeMarker, F: FnMut(&T, &T) -> bool>(
             right = right.add((!left_nonempty) as usize);
         }
 
-        // We now should have consumed the full input exactly once. This can
-        // only fail if the comparison operator fails to be Ord, in which case
-        // we will panic and never access the inconsistent state in dst.
+        // We now should have consumed the full input exactly once. This can only fail if the
+        // user-provided comparison operator fails implements a strict weak ordering as required by
+        // Ord incorrectly, in which case we will panic and never access the inconsistent state in
+        // dst.
```

Comment:
This involves repetition enough to get a double negation? Maybe triple, depending on how you scan it? It can't possibly literally mean "fails to implement Ord incorrectly"?

Also you later say "comparison function", and we should be consistent (unless this code is never hit by `sort_by`?)

```suggestion
        // user-provided comparison function fails to implement a strict weak ordering.
        // In that case we panic and never access the inconsistent state in dst.
```